app_flutter/lib/agent_dashboard_page.dart

```diff
@@ -0,0 +1,195 @@
+import 'navigation_drawer.dart';
+import 'sign_in_button.dart';
+import 'state/agent.dart';
+
+/// [AgentDashboardPage] parent widget that manages the state of the dashboard.
+class AgentDashboardPage extends StatefulWidget {
+  AgentDashboardPage({AgentState agentState})
+      : agentState = agentState ?? AgentState();
+
+  final AgentState agentState;
+
+  @visibleForTesting
+  static const Duration errorSnackbarDuration = Duration(seconds: 8);
+  @override
+  _AgentDashboardPageState createState() => _AgentDashboardPageState();
+}
+
+class _AgentDashboardPageState extends State<AgentDashboardPage> {
+  final GlobalKey<ScaffoldState> _scaffoldKey = GlobalKey<ScaffoldState>();
+
+  AgentState agentState;
+
+  @override
+  void initState() {
+    super.initState();
+
+    widget.agentState.startFetchingStateUpdates();
+
+    widget.agentState.errors.addListener(_showErrorSnackbar);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    agentState = widget.agentState;
+
+    return ChangeNotifierProvider<AgentState>(
+      create: (_) => agentState,
+      child: AgentDashboard(scaffoldKey: _scaffoldKey, agentState: agentState),
+    );
+  }
+
+  void _showErrorSnackbar() {
+    final Row snackbarContent = Row(
+      children: <Widget>[
+        const Icon(Icons.error),
+        const SizedBox(width: 10),
+        Text(agentState.errors.message)
+      ],
+    );
+    _scaffoldKey.currentState.showSnackBar(
+      SnackBar(
+        content: snackbarContent,
+        backgroundColor: Theme.of(context).errorColor,
+        duration: AgentDashboardPage.errorSnackbarDuration,
+      ),
+    );
+  }
+
+  @override
+  void dispose() {
+    agentState.errors.removeListener(_showErrorSnackbar);
+    super.dispose();
+  }
+}
+
+/// Shows current status of Flutter infra agents.
+class AgentDashboard extends StatefulWidget {
+  const AgentDashboard({this.scaffoldKey, this.agentState});
+
+  final GlobalKey<ScaffoldState> scaffoldKey;
+  final AgentState agentState;
+
+  @override
+  _AgentDashboardState createState() => _AgentDashboardState();
+}
+
+class _AgentDashboardState extends State<AgentDashboard> {
+  TextEditingController _agentIdController;
+  TextEditingController _agentCapabilitiesController;
+
+  @override
+  void initState() {
+    super.initState();
+
+    _agentIdController = TextEditingController();
+    _agentCapabilitiesController = TextEditingController();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Consumer<AgentState>(
+      builder: (_, AgentState agentState, Widget child) => Scaffold(
+        key: widget.scaffoldKey,
+        appBar: AppBar(
+          title: const Text('Infra Agents'),
+          actions: <Widget>[
+            SignInButton(authService: agentState.authService),
+          ],
+        ),
+        body: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            // Without this container the ListView will not be centered.
```

Comment:
Do you know why? If so could you comment here, or maybe do some digging - it seems odd at first glance

Reply:
No, originally the container had a height to make it look pretty, but when I went to remove it I noticed the issue. I think it has to do with the expanded, but I'll try to do some more digging around.